Pinta.Tools/Tools/TextTool.cs

```diff
@@ -598,33 +598,32 @@ private void UpdateMouseCursor()
 #endregion
 
 #region Keyboard Handlers
-		protected override void OnKeyDown (DrawingArea canvas, KeyPressEventArgs args)
+		protected override bool OnKeyDown (Document document, ToolKeyEventArgs e)
 		{
 			if (!PintaCore.Workspace.HasOpenDocuments) {
-				args.RetVal = false;
-				return;
+				return false;
 			}
 
-			Gdk.ModifierType modifier = args.Event.State;
+			//Gdk.ModifierType modifier = args.Event.State;
 
 			// If we are dragging the text, we
 			// aren't going to handle key presses
 			if (tracking)
-				return;
+				return false;
 
 			// Ignore anything with Alt pressed
-			if ((modifier & Gdk.ModifierType.Mod1Mask) != 0)
-				return;
+			if (e.IsAltPressed)
+				return false;
 
-			ctrlKey = (args.Event.Key == Gdk.Key.Control_L || args.Event.Key == Gdk.Key.Control_R);
-			UpdateMouseCursor ();
+			ctrlKey = (e.Key == Gdk.Key.Control_L || e.Key == Gdk.Key.Control_R);
```

Comment:
Should this have been using `IsControlPressed`?

Reply:
This one is technically different, as it is checking `args.Event.Key` instead of `args.Event.State`.  This means it is checking if `Control` is the pressed key that this event is for, rather than if `Control` is pressed along with another key(s).  It likely doesn't matter, but I refrained from making many changes to `TextTool` at this time as it's very complex and I haven't spent the time to fully understand it.

Reply:
Ah right, I'll make a note to look into this when I do the Cmd key changes for Mac